task/prefetch-dependencies/0.1/prefetch-dependencies.yaml

```diff
@@ -146,7 +146,27 @@ spec:
         update-ca-trust
       fi
 
-      cachi2 --log-level="$LOG_LEVEL" $config_flag fetch-deps \
+      cachi2 --log-level="$LOG_LEVEL" fetch-deps \
+      ACTIVATION_KEY_PATH="/activation-key"
+      if [ -d "$ACTIVATION_KEY_PATH" ]; then
+          echo "Detected activation key, registering with RHSM."
+        
+          # todo: make sure orgid key is consistent with buildah task and docs
+          subscription-manager register \
+            --org $(cat "/activation-key/orgid") \
+            --activationkey $(cat "/activation-key/activationkey")
+
+          # detect entitlement certs and setup environment variables
+          ls /etc/pki/entitlement/
+
+          export RHSM_ID=$(ls /etc/pki/entitlement/ | grep key | cut -d - -f 1)
+          echo $RHSM_ID
+          export C2_CLIENT_CERT="/etc/pki/entitlement/$RHSM_ID.pem"
+          export C2_CLIENT_KEY="/etc/pki/entitlement/$RHSM_ID-key.pem"
+          echo "Using client certificate $C2_CLIENT_CERT and key $C2_CLIENT_KEY."
+        fi
+
+      
```

Comment:
I don't see any unregistration here which is a good habit in general to do once we're done using a service to prevent a plethora of automated single-use registrations lingering in the subscription service.

Reply:
unregitration is now handled.